tcl/pd_canvaszoom.tcl

```diff
@@ -0,0 +1,333 @@
+package provide pd_canvaszoom 0.1
+
+namespace eval ::pd_canvaszoom:: {
+    namespace export zoominit
+    namespace export canvasxy
+    namespace export scalescript
+}
+
+proc ::pd_canvaszoom::zoominit {mytoplevel {zfact {1.1}}} {
+    set c [tkcanvas_name $mytoplevel]
+    # save zoom state in a global variable with the same name as the canvas handle
+    upvar #0 $c data
+    set data(zdepth) 1.0
+    set data(oldzdepth) 1.0
+    set data(idle) {}
```

Comment:
using a global variable with a common name (the canvas handle) looks a bit brittle.

i think it would be better to use a namespaced variable

Reply:
you're right, I lazily copied the code from https://wiki.tcl-lang.org/page/Canvas+zooming+using+mousewheel.
done now.